wpimath/src/main/java/edu/wpi/first/math/estimator/DifferentialDrivePoseEstimator.java

```diff
@@ -119,14 +190,32 @@ public DifferentialDrivePoseEstimator(
    *     theta]ᵀ, with units in meters and radians.
    */
   public void setVisionMeasurementStdDevs(Matrix<N3, N1> visionMeasurementStdDevs) {
-    var r = new double[3];
-    for (int i = 0; i < 3; ++i) {
+    setVisionMeasurementStdDevs3d(
+        VecBuilder.fill(
+            visionMeasurementStdDevs.get(0, 0),
+            visionMeasurementStdDevs.get(1, 0),
+            0,
+            visionMeasurementStdDevs.get(2, 0)));
+  }
+
+  /**
+   * Sets the pose estimator's trust of global measurements. This might be used to change trust in
+   * vision measurements after the autonomous period, or to change trust as distance to a vision
+   * target increases.
+   *
+   * @param visionMeasurementStdDevs Standard deviations of the vision measurements. Increase these
+   *     numbers to trust global measurements from vision less. This matrix is in the form [x, y,
+   *     theta]ᵀ, with units in meters and radians.
+   */
+  public void setVisionMeasurementStdDevs3d(Matrix<N4, N1> visionMeasurementStdDevs) {
```

Comment:
Why does this have four elements?

Reply:
I use the Rotation `times(double scalar)` function to scale the rotation component of the twist, but I could scale each component of the rvec individually I suppose. Would that be preferred over one scalar used for the rotation component?

Reply:
What you did makes sense. The docs for each element needs to be updated though, because it still says there's three elements.

Reply:
Did some additional reading and came across this - https://en.wikipedia.org/wiki/Rotation_formalisms_in_three_dimensions#Angle-Angle-Angle

This would seem to suggest we should go ahead with 3 orthogonal std dev inputs for rotation, because they're scaled as measurement errors on individual axes. This makes some sense; gyroscopes typically measure 3 independent axes so the input signals themselves are not coupled tightly, and could be different.

I'll update docs for the new functions to recognize std dev inputs as `x, y, z, roll, pitch, yaw`. Think I should change `heading` to `yaw` in the 2d function docs as well?

Reply:
Does scaling the twist elements individually work as expected? They aren't actually angles afterall, so they don't match the stddev units.

Reply:
I've decided to revert the change from earlier back to `[x, y, z, heading]` for stdev parameters + kalman gain after speaking with some people who know more about 3d rotation than I do. This hasn't yet led to the new 3d test passing, unfortunately.  

As per discussion on discord, it appears that `pose.exp(pose.log(other)) == other` doesn't appear to be held true within a reasonable margin of error - the translation component of the 
computed pose is always correct, but the heading differs significantly between the computed pose and the expected result. Data demonstrating this as attached.

[robot_vision_data.csv](https://github.com/wpilibsuite/allwpilib/files/10431925/robot_vision_data.csv)

columns are presented for the following:
- time of measurement within test
- robot_pose (sampled pose from pose buffer)
- vision_pose (input pose to `addVisionMeasurement` function)
- twist (`robot_pose.log(vision_pose)`)
- exp_pose ('robot_pose.exp(twist)`)
- pose_error (Translation3d between exp_pose and vision_pose)